### PR TITLE
[14.0][ADD] delivery_carrier_typology

### PR DESCRIPTION
**Pull request by chafique-delli**
This module allows:
- to specify the different means of transport used by a carrier (eg air freight, sea freight, etc.)
- to specify on a product, the means of transport which are prohibited (eg air freight for dangerous products)
- to select only delivery carriers via means of transport that are not prohibited for products in a sale order.

### CODE REVIEW

**Comment by rousseldenis** [comment]
@chafique-delli Thanks for this. Could you explain the difference with https://github.com/OCA/delivery-carrier/tree/14.0/delivery_carrier_category ?

**Comment by chafique-delli** [comment]
> @chafique-delli Thanks for this. Could you explain the difference with https://github.com/OCA/delivery-carrier/tree/14.0/delivery_carrier_category ?

@rousseldenis , I put a description of what the module does.

File: delivery_carrier_typology/models/delivery_carrier.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    _inherit = "delivery.carrier"
+
+    shipping_means_ids = fields.Many2many(
+        "delivery.carrier.typology",
+        string="Shipping means",
+    )
+
+
+class DeliveryCarrierTypology(models.Model):
+    _name = "delivery.carrier.typology"
```
Comment (rousseldenis) [inline delivery_carrier_typology/models/delivery_carrier.py:33]: @chafique-delli That should have been an exstension of carrier category no ?

Comment (chafique-delli) [reply delivery_carrier_typology/models/delivery_carrier.py:17]: yes, why not

Comment (chafique-delli) [reply delivery_carrier_typology/models/delivery_carrier.py:17]: @rousseldenis , According to you, the 'category_id' field would be used to indicate a means of transport used by the delivery carrier ?

Comment (rousseldenis) [reply delivery_carrier_typology/models/delivery_carrier.py:33]: Ok, it is more clear.

**Comment by rousseldenis** [comment]
@chafique-delli Could you check tests ?

**Comment by chafique-delli** [comment]
@rousseldenis , it's fixed.

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by chafique-delli** [comment]
@sebastienbeau 
@Kev-Roche

File: delivery_carrier_typology/models/product.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return res
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    prohibited_shipping_means_ids = fields.Many2many(
+        "delivery.carrier.typology",
+        string="Prohibited shipping means",
+    )
```
Comment (Kev-Roche) [inline delivery_carrier_typology/models/product.py:31]: If we create new variants after adding the prohibited shipping mean on the product template, the news variants will not have the prohibited shipping mean.

**Review by Kev-Roche — CHANGES_REQUESTED** [review]

File: delivery_carrier_typology/models/product.py (hunk 2 above)
Comment (chafique-delli) [reply delivery_carrier_typology/models/product.py:31]: @Kev-Roche , it's fixed.

**Comment by chafique-delli** [comment]
@Kev-Roche , the changes made, are ok for you?

**Pull request closed without merging**